src/main/java/stream/flarebot/flarebot/util/objects/expiringmap/ExpiringMap.java

```diff
@@ -31,60 +38,107 @@ public ExpiringMap(ExpiredEvent<K, V> expiredEvent) {
+            if (issueMS >= entry.getKey() || force) {
                 if (expiredEvent != null) {
-                    for (K k : a.getValue().getKey().keySet()) {
-                        expiredEvent.run(k, a.getValue().getKey().get(k), a.getKey(), a.getValue().getValue());
+                    for (K k : entry.getValue().getKey().keySet()) {
+                        expiredEvent.run(k, entry.getValue().getKey().get(k), entry.getKey(), entry.getValue().getValue());
                     }
                     if (!expiredEvent.isCancelled())
-                        e.remove();
+                        entryIterator.remove();
                 }
             } else
                 break;
         }
     }
 
+    /**
+     * Enters a specific key and value into the map with a specified time to expire after
+     *
+     * @param k             The key to enter into the map of type <K>.
+     * @param v             The value associated with the map to enter of type <V>.
+     * @param expireAfterMS The milliseconds after which the key-value will expire.
+     */
     public void put(K k, V v, long expireAfterMS) {
         long ms = System.currentTimeMillis() + expireAfterMS;
-        Pair<ConcurrentMap<K, V>, Long> c = elem.get(ms);
-        if (c == null) {
-            c = new Pair<>();
-            c.setKeyValue(new ConcurrentHashMap<>(), ms);
-            c.getKey().put(k, v);
-            elem.put(ms, c);
+        Pair<ConcurrentMap<K, V>, Long> pair = elem.get(ms);
+        if (pair == null) {
+            pair = new Pair<>();
+            pair.setKeyValue(new ConcurrentHashMap<>(), ms);
+            pair.getKey().put(k, v);
+            elem.put(ms, pair);
         } else {
-            c.getKey().put(k, v);
+            pair.getKey().put(k, v);
         }
     }
 
+    /**
+     * Checks whether the map contains a key-value mapping for the specified key.
+     *
+     * @param k The key to check existence of of type <K>
+     * @return Whether the key-value mapping is present in the map
+     */
     public boolean containsKey(K k) {
         return get(k) != null;
     }
 
+    @Deprecated
     public boolean containsValue(V v) {
         for (Pair<ConcurrentMap<K, V>, Long> pair : elem.values()) {
             return pair.getKey().containsValue(v);
         }
         return false;
     }
 
+    /**
+     * Returns the value for the key-value mapping indicated by the key.
+     * <p>
+     * If there is no key-value mapping for the key, the value returned is {@code null}.
+     *
+     * @param k The key to retrieve the value for.
+     * @return The value associated with the key.
+     */
     public V get(K k) {
         for (Pair<ConcurrentMap<K, V>, Long> pair : elem.values()) {
             if (pair.getKey().containsKey(k))
-                return pair.getKey().get(k);
+                pair.setValue(System.currentTimeMillis());
+            return pair.getKey().get(k);
```

Comment:
How come you return when it may not contain it? This will always stop on the first iteration

Reply:
Oh this was a mistake, will fix